gnu: Add legacy Zephyr compatibility

keith-packard opened the pull request:
In setup.sh, add option to create symlinks from gnu/ so that paths match previous SDK versions.

In Zephyr-sdkConfig.cmake, set ZEPHYR_TOOLCHAIN_VARIANT to zephyr/gnu only when Zephyr sets ZEPHYR_TOOLCHAIN_VARIANT_COMPILER_SUPPORT before finding the SDK module. This will require a change to Zephyr, which is unfortunate. Ideally, we'd use some existing mechanism to detect when to make this switch.

---

Fixes https://github.com/zephyrproject-rtos/sdk-ng/issues/851

stephanosio commented on scripts/template_setup_posix:
```diff
@@ -301,6 +307,14 @@ if [ "${do_cmake_pkg}" = "y" ]; then
+# Create compatibility links for old Zephyr versions
+if [ "${do_old_zephyr}" = "y" ]; then
+  echo "Creating links for old Zephyr compatibility ..."
+  ln -sr gnu/* . || assert_rc "ERROR: Creating toolchain links" 50
+  ln -sr cmake/zephyr/gnu/* cmake/zephyr || assert_rc "ERROR: Creating cmake links" 50
```
We also need to symlink `hosttools/*` to `.` for host tools.

keith-packard replied:
Thanks. done.

stephanosio reviewed (commented):
On second thought, "compatibility" is probably not the best choice of word here since SDK 1.0.0 is not meant to be "compatible" with Zephyr <= 4.2.

Maybe, s/compatibility/bisectability/ ?

keith-packard commented on scripts/template_setup_posix:
```diff
@@ -69,6 +69,7 @@ usage()
   echo "     all               Install all GNU toolchains"
   echo "  -l                   Install LLVM toolchain"
   echo "  -h                   Install host tools"
+  echo "  -o                   Old zephyr version compatibility"
```
Otherwise, yup, make it clear what this does.

keith-packard commented:
I added a commit which inverts the test for when to use -O2 with libstdc++ to preserve bisectability with the existing Zephyr bits. I also added a random cleanup commit which removes a stale comment in the Kconfig file.
